CMakeLists.txt

```diff
@@ -1,70 +1,34 @@
 cmake_minimum_required(VERSION 3.5)
-project(ursadb VERSION 1.3.0)
+project(ursadb VERSION 1.3.0
+               LANGUAGES CXX)
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+include(ClangFormat)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_FLAGS "-Wall")
-
-message("Looking for clang-format")
-find_program(CLANG_FORMAT_EXEC clang-format)
-if(NOT EXISTS ${CLANG_FORMAT_EXEC})
-    message(WARNING "clang-format not found! Target 'format' won't be available")
-else()
-    file(GLOB FORMAT_SOURCES *.cpp *.h)
-    add_custom_target(format
-      COMMAND
-       ${CLANG_FORMAT_EXEC}
-       -style=file
-       -i
-       ${FORMAT_SOURCES}
-      WORKING_DIRECTORY
-       ${CMAKE_SOURCE_DIR}
-      COMMENT
-      "Formatting..."
-    )
-endif()
-
-configure_file(Version.h.in Version.h)
 
-add_library(
-    ursa
-    STATIC
-    Utils.cpp
-    OnDiskDataset.cpp
-    OnDiskIterator.cpp
-    DatasetBuilder.cpp
-    Database.cpp
-    Trim.cpp
-    OnDiskIndex.cpp
-    FlatIndexBuilder.cpp
-    BitmapIndexBuilder.cpp
-    MemMap.cpp
-    Query.cpp
-    QueryParser.cpp
-    ExclusiveFile.cpp
-    DatabaseSnapshot.cpp
-    Task.cpp
-    Indexer.cpp
-    DatabaseHandle.cpp
-    ZHelpers.cpp
-    Responses.cpp
-    RawFile.cpp
-    OnDiskFileIndex.cpp
-)
-target_include_directories(ursa PRIVATE ${PROJECT_BINARY_DIR})
-target_link_libraries(ursa -lstdc++fs -lzmq -pthread)
+add_subdirectory(libursa)
 
-add_executable(ursadb Daemon.cpp NetworkService.cpp)
+add_executable(ursadb src/Daemon.cpp
+                      src/Daemon.h
```

Comment:
Should we add headers too? AFAIK usually X.cpp should always include X.h anyway.

Reply:
Actually, it's intended. CMake is smart and can figure it out itself, however it's meant for clangformat target.

From a few resources, I've found out that that globbing is a bad practice in cmake and adding them here explicitly will make those files discoverable for clang-format.